tools/scylla-nodetool.cc

```diff
@@ -1751,22 +1752,53 @@ void restore_operation(scylla_rest_client& client, const bpo::variables_map& vm)
         }
         params[required_param] = vm[required_param].as<sstring>();
     }
-    if (!vm.contains("sstables")) {
-      throw std::invalid_argument("missing required possitional argument: sstables");
+    bool sstables_as_params = vm.contains("sstables");
+    bool sstables_as_file_list = vm.contains("sstables-file-list");
+    if (not sstables_as_params and not sstables_as_file_list) {
+      throw std::invalid_argument("missing both argument: sstables and --sstables-file-list (at least one is required)");
     }
     if (vm.contains("scope")) {
         params["scope"] = vm["scope"].as<sstring>();
     }
-    sstring sstables_body = std::invoke([&vm] {
+
+    std::vector<sstring> sstables_list;
+    
+    // add the list given by the file param
+    if (sstables_as_file_list) {
+        sstring sstables_list_file = vm["sstables-file-list"].as<sstring>();
+        std::ifstream ifs(sstables_list_file);
+        if (ifs) {
+			std::string toc_txt;
+			while (std::getline(ifs, toc_txt)) {
+                if (not toc_txt.empty()) {
+                    sstables_list.push_back(toc_txt);
+                }
+			}
+        }
+        else {
+            throw std::invalid_argument(fmt::format("can't open: {}", sstables_list_file));
+        }
+    }
```

Comment:
please close the `ifs` somewhere

Reply:
I specifically verified it (see [destructor](https://en.cppreference.com/w/cpp/io/basic_ifstream.html)) since we might be throwing here.